### PR TITLE
fix: Comparison of optionals and type conversion clarification

### LINKED ISSUE #173: Fix implicit type conversions for `==` and `!=`
> Doesn't perform implicit type conversions

> Cell, implicitly compares via .hash()

> Slice, implicitly compares via .hash()

### LINKED ISSUE #174: Document optionals can be compared with `==` and `!=`

### PR DESCRIPTION
**Pull request by novusnota**
* Made comparison of optionals much more visible
* Clarified implicit conversions for Cells and Slices for == and !=
* A couple of fixes of past typos
* Many smaller enhancements

Closes #173 (clarification of statements on implicit comparison)
Closes #174 (better visibility of examples with comparing optionals)

### CODE REVIEW

**Review by anton-trunov — APPROVED** [review]

File: pages/book/operators.mdx [hunk 1]
```diff
@@ -287,16 +287,16 @@ Perform inequality and equality checks, find bigger, smaller or equal values.
 
 #### Not equal, `!=` [#binary-not-equal]
 
-Binary inequality (_not equal_) operator `!={:tact}` checks whether its two operands are _not equal_, returning a result of type [`Bool{:tact}`](/books/types#booleans). Doesn't perform implicit type conversions, always strict and requires both operands to be of the same type.
+Binary inequality (_not equal_) operator `!={:tact}` checks whether its two operands are _not equal_, returning a result of type [`Bool{:tact}`][bool]. Requires both operands to be of the same type and doesn't perform implicit type conversions, except for the [`Cell{:tact}`][p] and [`Slice{:tact}`][p] types, which are implicitly compared by their hashes.
 
-Can be applied to values of following types:
+Can be applied to [optionals](/book/optionals) and values of following types:
```
Comment (anton-trunov) [inline pages/book/operators.mdx:292]: I'd add `T?` to the list below (with the link to [optionals](https://docs.tact-lang.org/book/optionals)). It's easier to scan bullet points if you are using the book as a references

**Review by anton-trunov — CHANGES_REQUESTED** [review]
Let's fix just this one thing about `T?`

**Comment by anton-trunov** [comment]
Actually, this duplication of descriptions for `!=` and `==` should be fixed. I think it makes sense to merge these two subsections together

**Comment by novusnota** [comment]
> Actually, this duplication of descriptions for `!=` and `==` should be fixed. I think it makes sense to merge these two subsections together

Same can be said for `>` and `<`, `>=` and `<=`, `+` and `-`, etc. :)

**Comment by anton-trunov** [comment]
> Same can be said for > and <, >= and <=, + and -, etc. :)

not really -- those are a lot less polymorphic than `==` and `!=`

**Comment by anton-trunov** [comment]
basically, I'm not a fan of maintaining two non-trivial lists of types for `==` and `!=`

**Comment by novusnota** [comment]
* Applied the suggestion for `==` and `!=`
* Made the lists of available application types complete and exhaustive (although didn't use `T?`, but used "Optionals and `null` value" instead)

**Review by anton-trunov — APPROVED** [review]

**Pull request merged**